src/types.ts

```diff
@@ -13,6 +13,10 @@ export interface Task extends Assignment {
 	completed: string[]
 	confirmed: string[]
 	due: Timestamp
+	gcrCourseID?: string
```

Comment:
I think in general the typing of google classroom tasks could use some improvements, especially because you have to resort to using `any` many times in `ConfirmationTables.tsx`. We want to enforce that gcrCourseID, gcrID, gcrName, and gcrUserId are optional, but should all be included at once, correct? So we should make a new type `export interface GCRTask extends Task` that has these four properties as required. This new task should still work with all current task-related functionality because it is an extension of Task, but utilizing the `GCRTask` type should help you improve the typing in `CreateGCRTask`, `CreateTaskModal` (not having to pass in empty strings for these values), and `ConfirmationTables`

Reply:
see slack - would this just make every task a gcr task because i want to check in student and teacher modals whether it is a gcr task and indicate the name of the gcr task if so?